app/helpers/firmware_registry_helper/textual_summary.rb

```diff
@@ -36,4 +36,8 @@ def textual_firmware_binaries
   def firmware_binaries
     @record.firmware_binaries.order('name, version DESC').collect { |b| [textual_link(b), b.version, b.description[0..30]] }
   end
+
+  def textual_url
+    @record.endpoint&.url
```

Comment:
Should this be `@record&.endpoint&.url`?

I think there's no guarantee a FirmwareRegistry will have an endpoint.